File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int start = 0,end = numbers.size()-1;
        int sum=0;
        while(start<end){
            if(numbers[start]+numbers[end]>target){
                end--;
            }
            else if(numbers[start]+numbers[end]<target){
                start++;
            }
            else {
                vector<int>ans;
                ans.push_back(start+1);
                ans.push_back(end+1);
                return ans;
            }

        }
    }
};

File: Data Structures & Algorithms/combination-target-sum-ii/submission-0.py
class Solution:
    def combinationSum2(self, candidates: List[int], target: int) -> List[List[int]]:
        candidates.sort()
        final_ans=[]
        def ps(ans,nums,i,sum):
            if sum==target:
                final_ans.append(ans.copy())
                return
            if i==len(nums) or sum>target:
                return
            ans.append(nums[i])
            ps(ans,nums,i+1,sum+nums[i])
            ans.pop()
            j=i
            while j<len(nums) and nums[j]==nums[i]:
                j+=1
            ps(ans,nums,j,sum)
        ps([],candidates,0,0)
        return final_ans
            


        